File: android/HelloWorld/EditAndNormalize/src/main/java/com/dynamsoft/ddn/editandnormalize/MainViewModel.java
package com.dynamsoft.ddn.editandnormalize;

import androidx.lifecycle.ViewModel;

import com.dynamsoft.core.basic_structures.ImageData;
import com.dynamsoft.core.basic_structures.Quadrilateral;
import com.dynamsoft.dce.DrawingItem;
import com.dynamsoft.dce.QuadDrawingItem;

import java.util.ArrayList;

public class MainViewModel extends ViewModel {
    public ImageData normalizedImage;
    public ImageData capturedWholeImage;
    public Quadrilateral[] capturedQuads;

    public ArrayList<DrawingItem> getQuadDrawingItems() {
        if(capturedQuads == null) {
            return new ArrayList<>();
        }
        ArrayList<DrawingItem> items = new ArrayList<>();
        for (Quadrilateral quad : capturedQuads) {
            items.add(new QuadDrawingItem(quad));
        }
        return items;
    }
}

File: android/HelloWorld/EditAndNormalizeKt/src/main/java/com/dynamsoft/editandnormalizekt/MainViewModel.kt
package com.dynamsoft.editandnormalizekt

import androidx.lifecycle.ViewModel
import com.dynamsoft.core.basic_structures.ImageData
import com.dynamsoft.core.basic_structures.Quadrilateral
import com.dynamsoft.dce.DrawingItem
import com.dynamsoft.dce.QuadDrawingItem

class MainViewModel : ViewModel() {
    var normalizedImage: ImageData? = null
    var capturedWholeImage: ImageData? = null
    var capturedQuads: Array<Quadrilateral?>? = null;
    val quadDrawingItems: ArrayList<DrawingItem<*>>
        get() {
            if (capturedQuads == null) {
                return ArrayList()
            }
            val items = ArrayList<DrawingItem<*>>()
            for (quad in capturedQuads!!) {
                items.add(QuadDrawingItem(quad!!))
            }
            return items
        }
}
